source/core/DualTypeVisitor.cpp

```diff
@@ -129,23 +129,13 @@ namespace Vireo
         Boolean isTypeXIntrinsicClusterType = typeRefX->IsIntrinsicClusterDataType(&typeXName);
         Boolean isTypeYIntrinsicClusterType = typeRefY->IsIntrinsicClusterDataType(&typeYName);
         if (isTypeXIntrinsicClusterType && isTypeYIntrinsicClusterType) {
-            success = IntrinsicClustersCompatible(typeRefX, pDataX, typeRefY, pDataY, operation);
+            success = operation->AreIntrinsicClustersCompatible(typeRefX, typeRefY);
```

Comment:
Comparison of intrinsic clusters is now delegated to the operation. Equality treats ComplexSingle and ComplexDouble as  incompatible, but for conversion purposes, they are compatible.